api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java

```diff
@@ -2141,4 +2168,139 @@ public CohortDefinition getPatientsTransferedOutInLastHomeVisitCard() {
+            + "       AND o.value_coded = ${dispensation} "
+            + "       AND e.encounter_datetime = last_dispensation_type.encounter_datetime "
+            + " GROUP  BY p.patient_id";
+
+    StringSubstitutor stringSubstitutor = new StringSubstitutor(map);
+
+    cd.setQuery(stringSubstitutor.replace(query));
+
+    return cd;
+  }
+
+  /**
+   * <b>For 6 or more months of ARVs dispensed to active patients on ART: </b>
+   * <li>All active patients whose one of the MDCs is marked as “@param dispensationTypes” with
+   *     Estado do MDC as Iniciar (I) or Continuar (C) in the last Ficha Clinica with MDCs
+   *     registered
+   *
+   * @param dispensationTypes The List of dispensation types concepts
+   * @param states The list of MDC states
+   * @return {@link CohortDefinition}
+   */
+  public SqlCohortDefinition getPatientsWithTypeOfDispensationOnLastMdcRecord(
+      List<Integer> dispensationTypes, List<Integer> states) {
+
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName(
+        "All active patients whose one of the MDCs is marked as “DT OR DS OR DA” with Iniciar or Continuar in the last Ficha Clinica with MDC");
+    cd.addParameter(new Parameter("onOrBefore", "onOrBefore", Date.class));
+    cd.addParameter(new Parameter("location", "location", Location.class));
+
+    Map<String, String> map = new HashMap<>();
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId().toString());
+    map.put("1257", hivMetadata.getContinueRegimenConcept().getConceptId().toString());
+    map.put("1256", hivMetadata.getStartDrugs().getConceptId().toString());
+    map.put("23724", hivMetadata.getGaac().getConceptId().toString());
+    map.put("23730", hivMetadata.getQuarterlyDispensation().getConceptId().toString());
+    map.put("165179", hivMetadata.getDispensaComunitariaViaApeConcept().getConceptId().toString());
+    map.put(
+        "165315", hivMetadata.getDescentralizedArvDispensationConcept().getConceptId().toString());
+    map.put("23888", hivMetadata.getSemiannualDispensation().getConceptId().toString());
+    map.put("23729", hivMetadata.getRapidFlow().getConceptId().toString());
+    map.put("165174", hivMetadata.getLastRecordOfDispensingModeConcept().getConceptId().toString());
+    map.put("165322", hivMetadata.getMdcState().getConceptId().toString());
+    map.put("dispensationTypes", StringUtils.join(dispensationTypes, ","));
+    map.put("states", StringUtils.join(states, ","));
+
+    String query =
+        " SELECT p.patient_id   "
+            + "FROM   patient p   "
+            + "           INNER JOIN encounter e ON e.patient_id = p.patient_id   "
+            + "           INNER JOIN obs otype ON otype.encounter_id = e.encounter_id   "
+            + "           INNER JOIN obs ostate ON ostate.encounter_id = e.encounter_id   "
+            + "           INNER JOIN (    "
+            + "    SELECT p.patient_id, MAX(e.encounter_datetime) AS last_encounter   "
```

Comment:
Any reason for this max?

Reply:
The spec says **in the last Ficha Clinica with MDCs registered**